python/pyarrow/array.pxi

```diff
@@ -75,13 +97,43 @@ cdef class DataType:
         else:
             raise NotImplementedError(str(self))
 
+    property logical_type:
+
+        def __get__(self):
+            cdef Type type_id = self.type.id()
+            if type_id in _logical_type_map:
+                return _logical_type_map[type_id]
```

Comment:
I think this would best go in pandas_compat.py if possible (same with the dict above)